src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.Linux.cs

```diff
@@ -363,13 +363,13 @@ private void Initialize()
                 }
 
                 int fileDescriptor = Interop.open(GpioMemoryFilePath, FileOpenFlags.O_RDWR | FileOpenFlags.O_SYNC);
-                if (fileDescriptor < 0)
+                if (fileDescriptor == -1)
                 {
                     throw new IOException($"Error {Marshal.GetLastWin32Error()} initializing the Gpio driver.");
                 }
 
                 IntPtr mapPointer = Interop.mmap(IntPtr.Zero, Environment.SystemPageSize, (MemoryMappedProtections.PROT_READ | MemoryMappedProtections.PROT_WRITE), MemoryMappedFlags.MAP_SHARED, fileDescriptor, GpioRegisterOffset);
-                if (mapPointer.ToInt32() < 0)
+                if (mapPointer.ToInt32() == -1)
```

Comment:
Should you convert to Int64? IntPtr has different size on different platform and I'm not sure this check will be correct for say ARM64

Reply:
Well that's a good question.... This is code that is specific to the RaspberryPi3 (and now 4). and they are Arm32 and so I suppose they have a 32bit address space which would indicate that the conversion to Int32 is correct.

PR now amended.

Reply:
@Frankenslag actually RPi3 is 64-bit, only Rasbian is 32-bit. I got one RPi3 running on ARM64 Ubuntu 😄
You can use this image: https://ubuntu.com/download/iot/raspberry-pi-2-3 for 64-bit testing (or just try using ToInt64 and make the code work with 32-bit and assume it will work for 64-bit as well - a bit more risky but I guess it's ok considering RPi is mostly used with 32-bit)

Reply:
@krwq . I am happy to test for both ubuntu and Raspbian (I have spare SDCards). I can confirm that I have built a test nuget package and have used it with the Ssd1351 sample I was working on and this is working on the Pi4 now. So I guess that covers SPI also.

Do you want to merge the PR now so that others can get working with the Pi4 or do you want to wait until I have tested against Ubuntu (might take a day or 2). I assume you use the Arm64 dotnet core runtime package ?

Reply:
@Frankenslag, I'd prefer to wait until the testing is complete unless it makes it easier to test then I'm fine with merging earlier

Reply:
@Frankenslag merging this, let's do the testing and any extra fixes later

Reply:
https://github.com/dotnet/iot/issues/557